src/main/java/hexgui/gui/MessageDialogs.java

```diff
@@ -0,0 +1,390 @@
+            optionalMessage,
+            type,
+            JOptionPane.YES_NO_OPTION,
+            options,
+            defaultOption,
+            -1);
+    return (result == affirmativeOption);
+  }
+
+  public boolean showWarningQuestion(
+      Component parent,
+      String mainMessage,
+      String optionalMessage,
+      String destructiveOption,
+      boolean isCritical) {
+    return showWarningQuestion(
+        null, parent, mainMessage, optionalMessage, destructiveOption, isCritical);
+  }
+
+  public boolean showWarningQuestion(
+      String disableKey,
+      Component parent,
+      String mainMessage,
+      String optionalMessage,
+      String destructiveOption,
+      boolean isCritical) {
+    return showWarningQuestion(
+        disableKey, parent, mainMessage, optionalMessage, destructiveOption, "Cancel", isCritical);
+  }
+
+  /**
+   * Show warning message to confirm destructive actions.
+   *
+   * @return true, if destructive was chosen; false if cancel was chosen.
+   */
+  public boolean showWarningQuestion(
+      String disableKey,
+      Component parent,
+      String mainMessage,
+      String optionalMessage,
+      String destructiveOption,
+      String nonDestructiveOption,
+      boolean isCritical) {
+    if (checkDisabled(disableKey)) return true;
+    Object[] options = new Object[2];
+    if (Platform.isMac()) {
+      options[0] = nonDestructiveOption;
+      options[1] = destructiveOption;
+    } else {
+      options[0] = destructiveOption;
+      options[1] = nonDestructiveOption;
+    }
+    Object defaultOption = nonDestructiveOption;
+    int type;
+    if (isCritical) type = JOptionPane.WARNING_MESSAGE;
+    else type = JOptionPane.PLAIN_MESSAGE;
+    String title = "Warning" + " - " + m_applicationName;
+    Object result =
+        show(
+            disableKey,
+            parent,
+            title,
+            mainMessage,
+            optionalMessage,
+            type,
+            JOptionPane.YES_NO_OPTION,
+            options,
+            defaultOption,
+            -1);
+    return (result == destructiveOption);
+  }
+
+  private final String m_applicationName;
+
+  private final Set<String> m_disabled = new TreeSet<String>();
+
+  private static void addFiller(JComponent component) {
+    Box.Filler filler = GuiUtil.createFiller();
+    filler.setAlignmentX(Component.LEFT_ALIGNMENT);
+    component.add(filler);
+  }
+
+  private boolean checkDisabled(String disableKey) {
+    if (disableKey == null) return false;
+    Preferences prefs = PrefUtil.createNode("net/sf/hexgui/gui/messagedialogs/disabled");
+    boolean permanentlyDisabled = prefs.getBoolean(disableKey, false);
+    if (permanentlyDisabled) return true;
+    // Make sure this entry exists (right now these settings can only
+    // be directly edited in the backing store)
+    prefs.putBoolean(disableKey, permanentlyDisabled);
+    return m_disabled.contains(disableKey);
+  }
+
+  private Object show(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 371-388 into a separate method.